### PR TITLE
Graph cigar improvements

### PR DESCRIPTION
**Pull request by edotau**

### CODE REVIEW

**Review by cshoben — COMMENTED** [review]
The FaSeqToNode() really streamlines the cigarToGraph() function. Excited to see this still being worked on a bit! I had a comment on the test file, but everything else looks great to me!

File: cmd/globalAlignment/globalAlignment_test.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func compareEdges(manual *simpleGraph.SimpleGraph, graph *simpleGraph.SimpleGrap
 	prevNodeMatchCounter := 0
 	for i := 1; i < len(manual.Nodes); i++ {
 		for j := 0; j < len(manual.Nodes[i].Prev); j++ {
-			if manual.Nodes[i].Prev[j].Dest.Name == graph.Nodes[i].Prev[j].Dest.Name {
+			if len(manual.Nodes[i].Prev[j].Dest.Seq) == len(graph.Nodes[i].Prev[j].Dest.Seq) {
```
[the remark this answers is not in the document]
Comment (edotau) [reply cmd/globalAlignment/globalAlignment_test.go:101]: Yes you are correct, if we had written a more stringent test, we would have to come up with a better logic to check this.I think based on the context of how the test was written to compare to a manual graph, comparing the len isn't all too different from comparing the names. 

Most importantly, there is also a naming convention for naming the nodes that is incomplete. I added a TODO to address that and until that function is complete comparing node names will fail every test. Once that is written, what needs to be programmed is comparing two nodes and comparing their equality by checking (ID, Name, Sequence, Annotation) to determine if the nodes are the same.

File: simpleGraph/globalAlignment.go [hunk 2]
```diff
@@ -61,3 +64,21 @@ func NeedlemanWunsch(alpha []dna.Base, beta []dna.Base, scores [][]int64, gapPen
 	reverseCigarPointer(route)
 	return m[len(alpha)-1][len(beta)-1], route
 }
+
+// FaSeqToNode is a general function used create a new node based on a target fasta, query fasta and a cigar operation.
+// In addition, given two indices, it will update start/end for the subset of bases used to create the new Node.
+// TODO: Add logic for correct node name annotation convention
+func FaSeqToNode(target *fasta.Fasta, query *fasta.Fasta, tStart int, qStart int, cigar align.Cigar, index int) (*Node, int, int) {
```
[the remark this answers is not in the document]
Comment (edotau) [reply simpleGraph/globalAlignment.go:71]: Yes that is correct.

File: cmd/globalAlignment/globalAlignment_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cshoben) [reply cmd/globalAlignment/globalAlignment_test.go:101]: Thanks Eric! That makes a lot of sense. I hadn't thought about the code behind the name generation needing it's own tweaks.

[the remark this answers is not in the document]
Comment (cshoben) [reply cmd/globalAlignment/globalAlignment_test.go:101]: Also, my b for continuously opening a new review. The last comment was meant to just be a comment. :)

**Review by cshoben — DISMISSED** [review]
Looks good!

File: cmd/globalAlignment/globalAlignment_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (edotau) [reply cmd/globalAlignment/globalAlignment_test.go:101]: You are good, we are on the same page, I just wanted to expand on your comment!

File: simpleGraph/globalAlignment.go [hunk 3]
```diff
@@ -61,3 +64,21 @@ func NeedlemanWunsch(alpha []dna.Base, beta []dna.Base, scores [][]int64, gapPen
...
+	case 0:
+		curr := Node{Id: uint32(index), Name: fmt.Sprintf("%s_%d_%d_%s_%d_%d", target.Name, tStart, tStart+int(cigar.RunLength), query.Name, qStart, qStart+int(cigar.RunLength)), Seq: target.Seq[tStart : tStart+int(cigar.RunLength)]}
+		return &curr, tStart + int(cigar.RunLength), qStart + int(cigar.RunLength)
+	case 1:
+		ins := Node{Id: uint32(index), Name: fmt.Sprintf("%s_%d_%d_%s_%d_%d", target.Name, tStart, tStart, query.Name, qStart, qStart+int(cigar.RunLength)), Seq: query.Seq[qStart : qStart+int(cigar.RunLength)]}
+		return &ins, tStart, qStart + int(cigar.RunLength)
+	case 2:
+		del := Node{Id: uint32(index), Name: fmt.Sprintf("%s_%d_%d_%s_%d_%d", target.Name, tStart, tStart+int(cigar.RunLength), query.Name, qStart, qStart), Seq: target.Seq[tStart : tStart+int(cigar.RunLength)]}
+		return &del, tStart + int(cigar.RunLength), qStart
+	}
```
Comment (craiglowe) [inline simpleGraph/globalAlignment.go:82]: Should we have a default here that throws an error if we get an expected value of cigar.Op?

File: cmd/globalAlignment/globalAlignment_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (craiglowe) [reply cmd/globalAlignment/globalAlignment_test.go:101]: Could we call dna.CompareSeqsCaseSensitive(alpha []Base, beta []Base) instead of len()?

**Review by craiglowe — APPROVED** [review]
Looks great!

File: cmd/globalAlignment/globalAlignment_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (edotau) [reply cmd/globalAlignment/globalAlignment_test.go]: I had not thought of that. That is much better

File: simpleGraph/globalAlignment.go (hunk 3 above)
Comment (edotau) [reply simpleGraph/globalAlignment.go:84]: Yes I believe that is good to catch any additional errors. I also updated the cases to take the align package constants.

**Review by craiglowe — APPROVED** [review]
Looks great!

**Review by cfauci — APPROVED** [review]

**Review by DanielSnellings — APPROVED** [review]

**Pull request merged**